aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -1018,6 +1019,38 @@ Tensor& heaviside_(Tensor& self, const Tensor& values) {
   return at::heaviside_out(self, self, values);
 }
 
+Tensor& ldexp_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  TORCH_CHECK(!self.is_complex(), "ldexp does not support complex tensors.");
+  TORCH_CHECK(isIntegralType(other.scalar_type()), "ldexp requires integral types for second argument.");
+  Tensor other_;
+  if (self.dtype() == at::kFloat) {
+    other_ = other.to(at::kInt);
+  } else if (self.dtype() == at::kDouble) {
+    other_ = other.to(at::kLong);
+  } else {
+    TORCH_CHECK(false, "ldexp requires a float type and integral type of the same size");
```

Comment:
nit: It would be nice for the error message to be informative by including the dtypes of the inputs. For example,
```
TORCH_CHECK(false, "ldexp(tensor, other) requires a float type and integral type of the same size. Got dtype ", self.dtype(), " for `tensor` and ", other.dtype(), " for `other`");
```

Reply:
Can you elaborate on why this check is needed?

TensorIterator has its own logic for shape checking and broadcasting for binary ops. Could this broadcast like the other binary ops?

Reply:
"float type and integral type of the same size" refers to how the width of the floating-point type (e.g., fp32, has 32-bit width) should ideally match the width of the integral type (e.g., int32). 

The error message is not actually correct here, because this version of `ldexp` doesn't actually *enforce* that the `self` and `other` tensors passed by the user have types with the same width.

Reply:
Aha. Thanks @zou3519, I read that too quickly. Yeah this seems like an unnecessary requirement, especially if ldexp is implemented as `a * 2 ** b`.